### PR TITLE
setting up merged route /forecast/{gsp_id}/{only_values}

### LINKED ISSUE #131: API: Re define API routes
After reviewing, we shall go for this:

/v0/solar/GB/national/forecast/
/v0/solar/GB/national/pvlive/
/v0/solar/GB/gsp/forecast/all
/v0/solar/GB/gsp/forecast/{gsp_id}
/v0/solar/GB/gsp/forecast/{gsp_id}/{only_values} or other filter parameters
/v0/solar/GB/gsp/pvlive/{gsp_id}
/v0/system/GB/gsp/boundaries
/v0/system/GB/gsp/

TODOs (in the future)
/v0/wind/GB/
/v0/solar/site/
/v0/IT/

# Tasks

## National 
- Split national functions into separate file called 'national'.py' and rename routes ( + update tests). You need to add a route in 'main.py'. Around line 97 - `app.include_router(national_router, prefix=f"{v0_route}/national")`

'/forecast/national' --> 'forecast' 
'/pvlive/national' --> 'pvlive' 
new route prefix is `/v0/solar/GB/national`

## System

- Split system functions into separate file and rename routes ( + update tests)

- /v0/system/GB/gsp/boundaries
- /v0/system/GB/gsp/

new route prefix is `/v0/system/GB/`

## GSP

- Update gsp routes ( + updates tests)

'/forecast/all/' --> 'forecast/all'  (same)
'/forecast/one_gsp/{gps_id}' --> 'forecast/{gps_id}' 
'/pvlive/one_gsp/{gsp_id}' --> 'pvlive/{gsp_id}' 
same route prefix is `/v0/solar/GB/gsp`

### Extra merge
 - merge these routes

1. '/v0/GB/solar/gsp/forecast/latest/{gsp_id}'
2. '/v0/GB/solar/gsp//forecast/one_gsp/{gps_id}'

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

## Description

I've worked on adding a route to the gsp.py file that merges the routes get forecast for one gsp and get latest gsp. 

Fixes #131 task of merging the two routes. 

Image of updated routes: 

Screenshot 2022-09-15 at 13 21 57

Image of updated parameters for the merged route: 

Screenshot 2022-09-15 at 13 22 16

## How Has This Been Tested?

I visually checked that the route was added in the FastAPI interface, and also ran pytests locally. 

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/gsp.py [hunk 1]
```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
...
+#    logger.info(f"Get forecasts for gsp id {gsp_id} with {forecast_horizon_minutes=}")
+
+# # return get_latest_forecast_values_for_a_specific_gsp_from_database(
+#     session=session,
+#     gsp_id=gsp_id,
+#     forecast_horizon_minutes=forecast_horizon_minutes,
+# )
+
+
+@router.get("/forecast/{gsp_id}/{only_values}", response_model=Union[Forecast, List[Forecast]])
```
Comment (peterdudfield) [inline src/gsp.py:74]: looks good, might be worth moving this `only_values' to a input, but not in the URL. I think its an option but shouldnt always be stated. 

Could rename to only_forecast_values

File: src/gsp.py [hunk 2]
```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
...
+            historic=historic,
+        )
+
+        full_forecast.normalize()
+
+        return full_forecast
+
+    print("this is working")
+
+    only_values_forecast = get_latest_forecast_values_for_a_specific_gsp_from_database(
```
Comment (peterdudfield) [inline src/gsp.py:116]: goo to do and 
```
if :

else:
```

Just to make it really clear what is going on

File: src/gsp.py [hunk 3]
```diff
@@ -58,6 +58,70 @@ async def get_all_available_forecasts(
...
+            session=session,
+            gsp_id=gsp_id,
+            historic=historic,
+        )
+
+        full_forecast.normalize()
+
+        return full_forecast
+
+    print("this is working")
```
Comment (peterdudfield) [inline src/gsp.py:114]: Could change this to a logging statement

File: src/gsp.py (hunk 2 above)
Comment (rachel-labri-tipton) [reply src/gsp.py:116]: i had an else there but then deleted it. i'll put it back in.

Comment (peterdudfield) [reply src/gsp.py:116]: thanks

**Comment by rachel-labri-tipton** [comment]
@peterdudfield, all checks have passed! would you be able to have a quick look to see if I missed anything before I merge? 

I'm planning on making a new issue for updating the description for the merged route and could work on that tomorrow or later today.

File: src/gsp.py [hunk 4]
```diff
@@ -58,12 +58,14 @@ async def get_all_available_forecasts(
...
     gsp_id: int,
     session: Session = Depends(get_session),
     historic: Optional[bool] = False,
-) -> Forecast:
+    only_forecast_values: Optional[bool] = False,
+    forecast_horizon_minutes: Optional[int] = None,
+) -> Union[Forecast, List[ForecastValue]]:
     """### Get one forecast for a specific GSP
 
     The return object is a solar forecast with GSP system details.
```
Comment (peterdudfield) [inline src/gsp.py:71]: Would you be able to update this documentation. I.e also mention about just get the Forecast Values. Hopefully its just copying stuff from the old documentation

**Comment by peterdudfield** [comment]
Looks, great, just one thing to address. 

Could you also add a screenshot of the routes from the /docs page? You'll have to run it locally and got to the docs page.

File: src/gsp.py (hunk 4 above)
Comment (rachel-labri-tipton) [reply src/gsp.py:71]: I just made a new issue to update the description. probably isn't necessary? I'll go ahead and update the description on this pull request.

Comment (peterdudfield) [reply src/gsp.py:71]: yea, lets keep it together, if thats ok

**Comment by rachel-labri-tipton** [comment]
hey @peterdudfield I updated the route description and added screenshots to the original pull request. would you mind having a quick look at the new description?

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
@all-contributors please add @rachel-labri-tipton  for code

**Pull request merged**